Gems/ROS2/Code/Source/ROS2EditorSystemComponent.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+#include <AzCore/Serialization/SerializeContext.h>
+#include <ROS2EditorSystemComponent.h>
+
+namespace ROS2
+{
+    void ROS2EditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<ROS2EditorSystemComponent, ROS2SystemComponent>()
+                ->Version(0);
+        }
+    }
+
+    ROS2EditorSystemComponent::ROS2EditorSystemComponent() = default;
+
+    ROS2EditorSystemComponent::~ROS2EditorSystemComponent() = default;
```

Comment:
minor: You can probably just declare the default constructor/destructor in the header file only

Reply:
This is part of Gem template.
Also defining the default in the .cpp file allows the gem maintainers to update the System Components constructor/destructor without a potential longer compile if the system component header is included in multiple files

Reply:
Ok, that makes sense